RoddyCore/test/de/dkfz/roddy/config/ConfigurationValueTest.groovy

```diff
@@ -13,15 +13,137 @@ import org.junit.Test
+class ConfigurationValueTest {
 
-    // Test for determineTypeOfValue is in RoddyConversionHelperMethodsTest.groovy
-    // The method is directly bound to several methods of the conversion class and has nearly no
-    // logic of its own. So we put it there.
+    /**
+     * Given are two configurations:
+     * C extends B, B extends A
+     * =>   A -> B
+     *
+     * Configurations contain variables a and b
+     * A = { a = 'abc', b = 'abc${a}' }         <br/> Idea will try to pull A B C on the same line on code format.
+     * B = { a = 'def' }                        <br/>
+     * expected C.b = 'klm'
+     * expected B.b = 'hij'
+     */
+    @Test
+    void testCValueSubstitutionWithCfgDualChain() {
+        Configuration A = new Configuration(null)
+        Configuration B = new Configuration(null, A)
+
+        def cvA = A.getConfigurationValues()
+        def cvB = B.getConfigurationValues()
+
+        cvA.put('a', 'abc')
+        cvA.put('b', 'abc${a}')
+
+        cvB.put('a', 'def')
+
+        assert cvB['a'].value == 'def'
+        assert cvB['b'].value == 'abc${a}'
+        assert cvB['b'].toString() == 'abcdef'
+    }
+
+    /**
+     * Given are three configurations:
+     * C extends B, B extends A
+     * =>   A -> B -> C
+     *
+     * Configurations contain variables a and b
+     * A = { a = 'abc', b = 'def' }             <br/> Idea will try to pull A B C on the same line on code format.
+     * B = { a = 'hij', b = '$a' }              <br/>
+     * C = { a = 'klm' }                        <br/>
+     * expected C.b = 'klm'
+     * expected B.b = 'hij'
+     */
+    @Test
+    void testCValueSubstitutionWithCfgTripleChain() {
+        Configuration A = new Configuration(null)
+        Configuration B = new Configuration(null, A)
+        Configuration C = new Configuration(null, B)
+
+        def cvA = A.getConfigurationValues()
+        def cvB = B.getConfigurationValues()
+        def cvC = C.getConfigurationValues()
+
+        cvA.put('a', 'abc')
+        cvA.put('b', 'def')
+
+        cvB.put('a', 'hij')
+        cvB.put('b', '${a}')
+
+        cvC.put('a', 'klm')
 
+        assert cvB['a'].value == 'hij'
+        assert cvB['b'].value == '${a}'
+        assert cvB['b'].toString() == 'hij'
+        assert cvC['b'].toString() == 'klm'
+    }
+
+    /**
+     * Test a complex hierarchy
+     * =>   {{A1, A2} -> B1, B2} -> C1
+     *
+     * A1 = { a = 'A1.a', b = '${a}' }              <br/> Idea will try to pull A B C on the same line on code format.
+     * A2 = { a = 'A2.a', b = 'A2.b' }              <br/>
+     * B1 = { c = 'B1.c', b = '${a}' }              <br/>
+     * B2 = { a = '${b} + ${c}', b = 'B2.b' }       <br/>
+     * C1 = { c = 'C.c' }                           <br/>
+     *
+     * A1.b = 'A1.a'
+     * A2.b = 'A2.b'
+     *
+     * B1.a = 'A2.a'
+     * B1.b = 'A2.b'
+     * B1.d = 'A2.a'
+     *
+     * B2.a = 'B2.b + ${c}'
+     *
+     * C1.a = 'B2.b + C.c'
+     * C1.b = 'B2.b'
```

Comment:
This means, that the priority is `B2 > B1`, right?

BTW: Say, C1.b would evaluate to (=>) B1.b, then

 `C1.b => B1.b = "$a" => B2.a = "$b + $c"` => "$b + C.c"`

But the last `$b` again refers to C1.b, so there would be a loop!

We have no way to catch this, in the moment, right? I guess we need to make a check and write an error, if there is a cyclic definition.